src/axom/sidre/CMakeLists.txt

```diff
@@ -105,6 +110,7 @@ set(sidre_depends
 
 blt_list_append(TO sidre_depends ELEMENTS hdf5 IF HDF5_FOUND)
 blt_list_append(TO sidre_depends ELEMENTS sparsehash IF SPARSEHASH_FOUND)
+blt_list_append(TO sidre_depends ELEMENTS mfem IF MFEM_FOUND)
```

Comment:
This can be potentially problematic: I think `sidre` should only depend on `mfem` when `MFEM_FOUND AND AXOM_ENABLE_MFEM_SIDRE_DATACOLLECTION`